ProcessMaker/processmaker
# Implement checkTemplateAssetHashChanged Method to Prevent Media Folder ID Updates

## Issue & Reproduction Steps
This PR introduces the `checkTemplateAssetHashChanged` method to resolve a PLG-related issue where media Folder IDs in the `public/storage/inst1234` directory are updated on every run of the `php artisan processmaker:sync-guided-templates` command. This update prevents unnecessary Folder ID changes and database bloating by only updating the assets when changes are detected.

## Solution

- New Method: Implement `checkTemplateAssetHashChanged` to generate and compare asset hashes.
- Template Details Update: Store the generated asset hash within the `template_details` column.
- Media Handling: If the hash changes, run `clearCollectionMedia` to remove deleted images and prevent duplicate images.

## How to Test

**1. Initial Setup:**
- On a different branch, run `php artisan processmaker:sync-guided-templates`.
- Navigate to `public/storage/` in your instance and note the Folder IDs.

**2. Switch to the New Branch:**
- Update your instance to this branch.
- In your .env file, add `GUIDED_TEMPLATE_BRANCH=bugfix/FOUR-15426`.
- Run `php artisan optimize`.
- Run `php artisan processmaker:sync-guided-templates`.

**3.  Verification:**
- Navigate back to `public/storage/`.
- Ensure the Folder IDs have not changed if no asset changes were made.
- Verify that Folder IDs only change if new images are added to the templates' asset directory.


## Related Tickets & Packages
-[ FOUR-15426](https://processmaker.atlassian.net/browse/FOUR-15426)
-[ wizard-templates PR](https://github.com/ProcessMaker/wizard-templates/pull/112)

ci:next

## Code Review Checklist
